server/src/lib/model/user-client.ts

```diff
@@ -245,17 +251,42 @@ const UserClient = {
     return true;
   },
 
-  // update the challenge_team of the user who is already signed up but have not joined any team
-  // [BUG]: there are exceptions if the team doesn't exist
+  // enroll an unenrolled but registered user
+  // [BUG]: there are exceptions if the team or challenge doesn't exist
   // TODO(riley): Hook this up to a constants file.
-  async updateChallengeTeam(
+  async enrollRegisteredUser(
     email: string,
-    challenge_team: string
+    challenge: string,
+    team: string,
+    invite: string
   ): Promise<boolean> {
-    return await db.query(
-      'UPDATE user_clients SET challenge_team = ? WHERE email = ? AND has_login AND challenge_team IS NULL',
-      [challenge_team, email]
-    );
+    if (email && challenge && team) {
+      const client_id = (await Promise.all([
+        UserClient.findClientId(email),
+      ]))[0];
+      if (client_id) {
+        // for user invitation, invite_by is the same value of url_token from another row
+        // for team invitation, invite_by is null
```

Comment:
The language of "team invitation" confused me here. The URL will either look like:

`?challenge=pilot&team=sap&invite=<uuid>` or,

`?challenge=pilot&team=sap`

So I think it's either a user invitation or no user invitation. Adding the "team invitation" comment sounds to me like teams have their own invite codes. I think you can just delete that second line.

---

Also, nitpick but there's a typo in the comment – it should be `invited_by`.

Reply:
(Fixed upstream)